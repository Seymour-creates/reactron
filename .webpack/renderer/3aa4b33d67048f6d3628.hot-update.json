{"h":"973fd84ce8468e13f3f0","c":{"main_window":true}}