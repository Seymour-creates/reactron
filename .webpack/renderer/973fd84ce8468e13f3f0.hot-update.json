{"h":"e77e8ae52fd264b48d25","c":{"main_window":true}}