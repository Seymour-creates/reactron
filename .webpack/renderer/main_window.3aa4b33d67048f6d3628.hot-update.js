webpackHotUpdate("main_window",{

/***/ "./src/components/ComponentTree.tsx":
false,

/***/ "./src/components/HeadNode.tsx":
false,

/***/ "./src/components/Header.tsx":
false,

/***/ "./src/components/Node.tsx":
false,

/***/ "./src/components/ReactComponent.tsx":
false,

/***/ "./src/components/RenderedContainer.tsx":
false,

/***/ "./src/components/RenderedPage.tsx":
/*!*****************************************!*\
  !*** ./src/components/RenderedPage.tsx ***!
  \*****************************************/
/*! no static exports found */
/***/ (function(module, exports) {

throw new Error("Module parse failed: Unexpected token (37:4)\nFile was processed with these loaders:\n * ./node_modules/ts-loader/index.js\nYou may need an additional loader to handle the result of these loaders.\n|                             react_1.default.createElement(Visualizer_1.default, null))));\n|     div >\n>     ;\n|     ;\n| }");

/***/ }),

/***/ "./src/components/StateContainer.tsx":
false,

/***/ "./src/components/StateItem.tsx":
false,

/***/ "./src/components/Visualizer.tsx":
false,

/***/ "path":
false

})
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IiIsImZpbGUiOiJtYWluX3dpbmRvdy4zYWE0YjMzZDY3MDQ4ZjZkMzYyOC5ob3QtdXBkYXRlLmpzIiwic291cmNlUm9vdCI6IiJ9