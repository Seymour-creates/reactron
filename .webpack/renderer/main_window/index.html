<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <title>ReacTron</title>
  </head>
  <div id="root"></div>
</html>

<script src="/main_window/index.js"></script>