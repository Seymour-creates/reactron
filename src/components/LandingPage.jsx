import React from 'react';
import { useState } from 'react';
import filesysHelpers from '../../filesysHelpers.js';
import axios from 'axios';

// Will house the landing page / initial render page
// Will import files from here

export default function LandingPage(props) {
  const [staticFile, useStaticFile] = useState();

  const [components, useComponents] = useState();

  const staticOnClick = async () => {
    const fileHandle = window.showOpenFilePicker();
    fileHandle
      .then((data) => data[0].getFile())
      .then((res) => res.text())
      .then((data) => {
        axios.post('/fs/stylesheet', { item: data });
        useStaticFile(fileHandle);
      });
  };

  const componentOnClick = async () => {
    const fileHandle = await window.showDirectoryPicker();
    const result = await filesysHelpers.directoryLogger(fileHandle);
    useComponents(result);
  };

  const submitDirs = async () => {
    filesysHelpers
      .fileDisplay(components, 'componentFiles')
      .then(async (data) => {
        const tempArr = [];
        const nameArr = [];
        data.forEach(async (elem) => {
          nameArr.push(elem.getFile().then((res) => res.name));
          tempArr.push(elem.getFile().then((res) => res.text()));
        });
        const fileContents = await Promise.all(tempArr);
        const nameContents = await Promise.all(nameArr);

        const resultArr = [];

        for (let i = 0; i < fileContents.length; i += 1) {
          // when user authentication is implemeneted, file name needs to be updated:
          // 'username'/'projectname'/'filename'
          resultArr.push({
            name: nameContents[i],
            contents: fileContents[i],
          });
        }

        props.useFilesArr(resultArr);

        axios.post('/fs/upload', {
          files: resultArr,
          username: 'sample',
          project: 'sampleApp',
        });

        props.useLoadStatus(true); //calls useloadStatus to change state to true
      });
  };

  return (
    <div className='landingPage' data-testid="LandingPage" >
      <div className='header'>
        <h1>Reactron</h1>
      </div>
      <div className='instructions'>
        <p>
          In order for Reactron to process your application files correctly,
          please follow these instructions. <br />
          If you have a CSS or SCSS file you would like processed, please upload
          it under the Static Directory. <br />
          Reactron will look for an <b>index.js</b> file that connects to an{' '}
          <b>App.jsx</b> component. Please upload your index.js, App.jsx, and
          any other component files in one directory under the Component
          Directory below.
        </p>
      </div>
      <div className='staticInstr'>
        <p>
          <b>StaticDirectory</b>
          <br /> <i>Example</i>
          <br /> - style.css
          <br /> <b>or</b>
          <br /> - style.scss
          <br />
          <br />
        </p>
      </div>
      <div className='componentInstr'>
        <p>
          <b>Component Directory</b>
          <br /> <i>Example</i>
          <br /> - index.js
          <br /> - App.jsx
          <br /> - Component1.jsx
          <br /> - Component2.jsx
        </p>
      </div>
      <div className='staticFiles'>
        <b>Static Files</b>
        <p>Please upload your static directory here.</p>
        <p>{staticFile ? `The  directory has been uploaded` : ''}</p>
        <button className='button' id='static' onClick={staticOnClick}>
          Select File Here
        </button>
      </div>
      <div className='componentFiles'>
        <b>Component Files</b>
        <p>Please upload your component files here.</p>
        <p>
          {components
            ? `The ${
                components[Object.keys(components)[0]].handle.name
              } directory has been uploaded`
            : ''}
        </p>
        <button className='button' id='component' onClick={componentOnClick}>
          Select File Here
        </button>
      </div>
      {staticFile && components ? (
        <div className='next'>
          <p>
            Your files have been successfully uploaded. Hit the next button for
            Reactron to begin the rendering process.
            <br />
            <button className='button' onClick={submitDirs}>
              Next
            </button>
          </p>
        </div>
      ) : (
        ''
      )}
    </div>
  );
}
