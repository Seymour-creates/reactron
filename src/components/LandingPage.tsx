import React from 'react';
import LandingDisplay from './LandingDisplay';

export default function LandingPage() {
  return <LandingDisplay />;
}
