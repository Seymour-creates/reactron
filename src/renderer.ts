/**
 * This file will automatically be loaded by webpack and run in the "renderer" context.
 * To learn more about the differences between the "main" and the "renderer" context in
 * Electron, visit:
 *
 * https://electronjs.org/docs/tutorial/application-architecture#main-and-renderer-processes
 *
 * By default, Node.js integration in this file is disabled. When enabling Node.js integration
 * in a renderer process, please be aware of potential security implications. You can read
 * more about security risks here:
 *
 * https://electronjs.org/docs/tutorial/security
 *
 * To enable Node.js integration in this file, open up `main.js` and enable the `nodeIntegration`
 * flag:
 *
 * ```
 *  // Create the browser window.
 *  mainWindow = new BrowserWindow({
 *    width: 800,
 *    height: 600,
 *    webPreferences: {
 *      nodeIntegration: true
 *    }
 *  });
 * ```
 */

// renderer is needed to render files to react for electron
// import files here
import './index.scss';
import './main';

// IPC stands for inter-process communication
// ipcRenderer allows async rendering, listens from main processor
const { remote, ipcRenderer } = require('electron');

// ipcMain, event emitter, allows async main processing, will listen from the renderer

const mainProcess = remote.require('./index');

// selectMe button in RenderedPage
const testVar = document.getElementById('selectMe');
// onClick, populate the page with file contents
testVar.addEventListener('click', () => {
  mainProcess.testFile();
});

// ipcRenderer.on('file-opened', (event, ...args) => {
//   console.log(args)
// })

console.log(
  'ğŸ‘‹ This message is being logged by "renderer.js", included via webpack'
);