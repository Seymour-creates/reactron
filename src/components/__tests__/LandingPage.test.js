import {render, cleanup, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import LandingPage from '../LandingPage';

afterEach(() => {
  cleanup();
});

test('test', () => {
  expect(true).toBe(true);
});

test('should render LandingPage', () => {
  render(<LandingPage />);
  const landing = screen.getByTestId('LandingPage');
  expect(landing).toBeInTheDocument();
  expect(landing).toContainHTML('div');
  expect(landing).toContainHTML('p');
  expect(landing).toContainHTML('h1');
  expect(landing).toContainHTML('button');
  expect(landing).toHaveTextContent('Reactron');
  expect(landing).toHaveTextContent('Static Directory');
  expect(landing).toHaveTextContent('Component Directory');
});

test('should match LandingPage snapshot', () => {
  const snap = renderer.create(<LandingPage />).toJSON();
  expect(snap).toMatchSnapshot();
});
