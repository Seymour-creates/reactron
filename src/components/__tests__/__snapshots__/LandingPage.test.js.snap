// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match LandingPage snapshot 1`] = `
<div
  className="landingPage"
  data-testid="LandingPage"
>
  <div
    className="header"
  >
    <h1>
      Reactron
    </h1>
  </div>
  <div
    className="instructions"
  >
    <p>
      In order for Reactron to process your application files correctly, please follow these instructions. 
      <br />
      If you have a CSS or SCSS file you would like processed, please upload it under the Static Directory. 
      <br />
      Reactron will look for an 
      <b>
        index.js
      </b>
       file that connects to an
       
      <b>
        App.jsx
      </b>
       component. Please upload your index.js, App.jsx, and any other component files in one directory under the Component Directory below.
    </p>
  </div>
  <div
    className="staticInstr"
  >
    <p>
      <b>
        StaticDirectory
      </b>
      <br />
       
      <i>
        Example
      </i>
      <br />
       - style.css
      <br />
       
      <b>
        or
      </b>
      <br />
       - style.scss
      <br />
      <br />
    </p>
  </div>
  <div
    className="componentInstr"
  >
    <p>
      <b>
        Component Directory
      </b>
      <br />
       
      <i>
        Example
      </i>
      <br />
       - index.js
      <br />
       - App.jsx
      <br />
       - Component1.jsx
      <br />
       - Component2.jsx
    </p>
  </div>
  <div
    className="staticFiles"
  >
    <b>
      Static Files
    </b>
    <p>
      Please upload your static directory here.
    </p>
    <p>
      
    </p>
    <button
      className="button"
      id="static"
      onClick={[Function]}
    >
      Select File Here
    </button>
  </div>
  <div
    className="componentFiles"
  >
    <b>
      Component Files
    </b>
    <p>
      Please upload your component files here.
    </p>
    <p>
      
    </p>
    <button
      className="button"
      id="component"
      onClick={[Function]}
    >
      Select File Here
    </button>
  </div>
  
</div>
`;
